src/_guides/language/sound-dart.md

```diff
@@ -348,6 +354,56 @@ and generic type arguments.
 
 When the analyzer can't infer the type, the `dynamic` type is assigned.
 
+How does type inference work with collections and generics?
```

Comment:
Seems like a lot of questions in a row... Also, it's unclear exactly what "following snippets" refers to.